### PR TITLE
feat(maven): bumpVersion support

### PR DESCRIPTION
**Pull request by kajstrom**
## Changes:

Adds bumpVersion option support for Maven manager.

## Context:

Relates to issue #9257 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

PR created as part of testing on a real repository: https://github.com/kajstrom/renovate-pom-bumpversion-test/pull/2/commits/7236981968e739f431bce85bde2c8185407d8779

### CODE REVIEW

**Review by JamieMagee — DISMISSED** [review]

File: lib/config/options/index.ts [hunk 1]
```diff
@@ -1194,7 +1194,7 @@ const options: RenovateOptions[] = [
     description: 'Bump the version in the package file being updated.',
     type: 'string',
     allowedValues: ['major', 'minor', 'patch'],
-    supportedManagers: ['helmv3', 'npm', 'sbt'],
+    supportedManagers: ['helmv3', 'npm', 'sbt', 'maven'],
```
Comment (astellingwerf) [inline lib/config/options/index.ts:1197]: I'd prefer to keep them in alphabetical order.
```suggestion
    supportedManagers: ['helmv3', 'npm', 'maven', 'sbt'],
```

File: lib/manager/maven/index.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import * as datasourceMaven from '../../datasource/maven';
 import * as mavenVersioning from '../../versioning/maven';
 
 export { extractAllPackageFiles } from './extract';
-export { updateDependency } from './update';
+export { updateDependency, bumpPackageVersion } from './update';
```
Comment (astellingwerf) [inline lib/manager/maven/index.ts:6]: This snippet violates ESLint `sort-imports`
```suggestion
export { bumpPackageVersion, updateDependency } from './update';
```

**Review by astellingwerf — CHANGES_REQUESTED** [review]
I noticed a few lines that can do with a bit of extra attention.

File: lib/manager/maven/update.ts [hunk 3]
```diff
@@ -44,3 +51,61 @@ export function updateDependency({
...
+    'Checking if we should bump pom.xml version'
+  );
+  let bumpedContent = content;
+
+  if (!currentValue) {
+    logger.warn('Unable to bump pom.xml version, pom.xml has no version');
+    return { bumpedContent };
+  }
+
+  if (!semver.valid(currentValue)) {
```
Comment (viceice) [inline lib/manager/maven/update.ts:71]: Are you sure this must be semver and not maven versioning?

Comment (astellingwerf) [reply lib/manager/maven/update.ts:71]: @viceice, are you sure it's maven versioning and not the versioning configured for this package?

File: docs/usage/configuration-options.md [hunk 4]
```diff
@@ -316,7 +316,7 @@ This is an advance field and it's recommend you seek a config review before appl
 
 ## bumpVersion
 
-Currently this setting supports `helmv3`, `npm` and `sbt` only, so raise a feature request if you have a use for it with other package managers.
+Currently this setting supports `helmv3`, `npm`, `sbt` and `maven` only, so raise a feature request if you have a use for it with other package managers.
```
Comment (astellingwerf) [inline docs/usage/configuration-options.md:319]:
```suggestion
Currently this setting supports `helmv3`, `npm`, `maven` and `sbt` only, so raise a feature request if you have a use for it with other package managers.
```

File: lib/manager/maven/update.ts (hunk 3 above)
Comment (kajstrom) [reply lib/manager/maven/update.ts:71]: @viceice @astellingwerf This is the package version being bumped. Followed along the lines of other packet managers here that the bumped version must be a semver.

File: docs/usage/configuration-options.md [hunk 5]
```diff
@@ -316,7 +316,7 @@ This is an advance field and it's recommend you seek a config review before appl
...
 Its purpose is if you want Renovate to update the `version` field within your file's `package.json` any time it updates dependencies within.
```
[the remark this answers is not in the document]
Comment (kajstrom) [reply docs/usage/configuration-options.md]: Changed to more generic "package file" term here. Later example mentions `package.json`, but I think it still works as good clarification in the documentation.

**Review by viceice — APPROVED** [review]

**Pull request merged**